ui/packages/catalog-ui-search/src/main/webapp/component/query-add/query-add.view.js

```diff
@@ -44,12 +45,25 @@ module.exports = Marionette.LayoutView.extend({
     'click .editor-saveRun': 'saveRun',
   },
   initialize: function() {
+    this.listenTo(user.getQuerySettings(), 'change', querySettings =>
+      this.updateCurrentQuery(querySettings)
+    )
     this.model = new Query.Model(this.getDefaultQuery())
     this.listenTo(this.model, 'resetToDefaults change:type', this.reshow)
     this.listenTo(this.model, 'change:filterTree', this.reshow)
     this.listenTo(this.model, 'closeDropdown', this.closeDropdown)
     this.listenForSave()
   },
+  updateCurrentQuery: function(currentQuerySettings) {
+    if (currentQuerySettings.get('type') === 'custom') {
```

Comment:
❓ Looks like this will only update the query when the `type` of the current query settings is `custom`. Is this expected behavior?

For instance: if I was on a `custom` search form and reset it to `default`, I don't think the `default` search form would appear in this case.

Note: after looking at it more, it looks like we expect the search form to *only* change to custom.

Reply:
We don't want this function to trigger for other changes to type in the model, like https://github.com/codice/ddf/blob/master/ui/packages/catalog-ui-search/src/main/webapp/component/search-interactions/search-interactions.view.js#L85. I'll have to look into the second part of your comment